File: 0001_0100/0058/0058.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    while n:
        n -= 1
        s = "YES\n"
        a, b = (int(x) for x in f.readline().split())
        p1 = [int(x) for x in f.readline().split()]
        for i in range(a - 1):
            p2 = [int(x) for x in f.readline().split()]
            for j in range(b - 1):
                if ((sum(p1[j : j + 2]) + sum(p2[j : j + 2])) in (0, 4)):
                    s = "NO\n"
                    break
            p1 = p2
        q.write(s)

        
    
           
        

File: 0001_0100/0058/0058.go
package main

import (
	"fmt"
	"os"
)

func readline(f *os.File, n int) []int8 {
	p := make([]int8, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(f, &p[i])
	}
	return p
}

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, a, b int
	fmt.Fscan(f, &n)
	for ; n > 0; n-- {
		s := "YES"
		fmt.Fscan(f, &a, &b)
		p1 := readline(f, b)
		for i := 1; i < a; i++ {
			p2 := readline(f, b)
			for j := 1; j < b; j++ {
				if p2[j]+p2[j-1]+p1[j]+p1[j-1] == 0 || p2[j]+p2[j-1]+p1[j]+p1[j-1] == 4 {
					s = "NO"
					break
				}
			}
			p1 = p2
		}
		fmt.Fprintln(q, s)
	}
}
